### DISCUSSION TITLE
What's the difference between the different version tags on the Kindle build environment docker? 0.2.3 vs 0.3.0

### DISCUSSION THREAD

**Discussion by melyux** [koreader/koreader#9923] [Q&A]
The ones [here](https://hub.docker.com/r/koreader/kokindle/tags). What are these the versions of?

**Comment by Frenzie** [comment]
I'm not entirely sure what you're asking. They're versions of the image itself. ;-P

In this case 0.3.0 mostly means a much newer CMake version.
https://github.com/koreader/virdevenv/pull/80

Everything's kind of stuck until I find the time to beat the incredibly shitty Android toolchain into submission. ;-)

**Reply by melyux** [reply]
I saw that the 0.2.3 tag was updated after the 0.3.0 tag, so I didn't know what to think :P

I ask because I'm running the docker `kokindle` image to build using `./kodev release kindle`, but the 0.3.0 version fails with this error every time

```
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/10/libgcc.a when searching for -lgcc
/usr/bin/ld: cannot find -lgcc
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/10/libgcc.a when searching for -lgcc
/usr/bin/ld: cannot find -lgcc
```

while the 0.2.3 version works perfectly. Is this part of the stuckness around the Android toolchain shenanigans?

**Reply by Frenzie** [reply]
> I saw that the 0.2.3 tag was updated after the 0.3.0 tag, so I didn't know what to think :P

Seems incredibly natural to me. /shrug

0.2.2 updated after 2.2.3 would be a touch stranger, but regardless, what you see on Docker Hub is just when it was pushed for whatever reason.

> Is this part of the stuckness around the Android toolchain shenanigans?

There's no such thing as a Kindle 0.3.0 image. Docker Hub is not a source of truth. ;-)

**Reply by Frenzie** [reply]
That is, if there's a 0.3.0 it's an unfinished thought or an early prototype.

**Reply by Frenzie** [reply]
Here's what's actively used by us:
https://gitlab.com/koreader/nightly-builds/-/blob/7749e79251dc5dd8f4bf8fcd57ad7ef2c0a5f3fe/.gitlab-ci.yml
https://github.com/koreader/koreader-base/blob/66a793438082fc8ba37667e05a38882cb78d9701/.circleci/config.yml

**Reply by Frenzie** [reply]
Note that this already includes everything from the linked PR *except* Android. That's how it's stuck.
